File: Practice/15/Python/sluchChislo.py
import random
print("Попробуйте угадать мое загаданное число, у вас 5 попыток\n")
randomN=random.randint(0, 100)
game = 1
win = 0
fails= 0
while game == 1:
    while fails < 5:
        userN=int(input())
        if userN>100 or userN<0:
            print("Введите число из диапазона(0-100)")
        elif userN>randomN:
            print("Загаданное число меньше")
            fails=fails+1
        elif userN<randomN:
            print("Загаданное число больше")
            fails=fails+1
        elif userN==randomN:
            print("Поздравляю, вы угадали число!")
            win=win+1
            fails=5
    if win == 0:
        if fails == 5:
            print("Вы проиграли, загаданное число:", randomN)
            print("Хотите сыграть снова?\n1.Да\n2.Нет\n")
            game = int(input())
            if game == 1:
                fails = 0
                print("Попробуйте угадать мое загаданное число, у вас 5 попыток\n")
                continue
            else:
                break
    if win == 1:
        print("Хотите сыграть снова?\n1.Да\n2.Нет\n")
        game = int(input())
        if game == 1:
            fails =0
            print("Попробуйте угадать мое загаданное число, у вас 5 попыток\n")
            continue
        else:
            break

File: Practice/15/C++/sluchChislo.cpp
#include<iostream>;
#include <ctime>;
using namespace std;
int randomN,userN,fails,win,game;
int main() {
	setlocale(LC_CTYPE, "rus");
	srand(time(0));
	randomN = 1 + rand() % 100;
	cout << "Попробуйте угадать мое загаданное число, у вас 5 попыток\n";
	fails = 0;
	game = 1;
	win = 1;
	while (game == 1) {
		while (fails < 5) {
			cin >> userN;
			if (userN > 100 or userN < 0) {
				cout << "Введите число в верном диапазоне\n";
			}
			else if (userN > randomN) {
				cout << "Загаданое число меньше\n";
				fails++;
			}
			else if (userN < randomN) {
				cout << "Загаданое число больше\n";
				fails++;
			}
			else if (userN == randomN) {
				win++; cout << "Поздравляю, вы угадали число!\n";
				fails = 5;
			}
		}
		if (win == 0) {
			if (fails == 5) {
				cout << "Вы проиграли\n";
				cout << "Хотите снова?\n1.Да\n2.Нет\n";
				cin >> game;
				if (game == 1) { fails = 0; 
				cout << "Попробуйте угадать мое загаданное число, у вас 5 попыток\n";
				}
				else break;
			}
		}
		if (win == 1) {
			cout << "Хотите снова?\n1.Да\n2.Нет\n";
			cin >> game;
			if (game == 1) { fails = 0;
			cout << "Попробуйте угадать мое загаданное число, у вас 5 попыток\n";
			}
			else break;
		}

	}
}